model-optimizer/extensions/ops/gathernd.py

```diff
@@ -0,0 +1,103 @@
+"""
+ Copyright (C) 2018-2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import numpy as np
+
+from mo.front.common.partial_infer.utils import int64_array
+from mo.graph.graph import Node, Graph
+from mo.ops.op import Op
+
+
+class GatherND(Op):
+    op = 'GatherND'
+
+    def __init__(self, graph: Graph, attrs: dict):
+        mandatory_props = {
+            'kind': 'op',
+            'type': self.op,
+            'op': self.op,
+            'version': 'opset5',
+            'infer': self.infer,
+            'in_ports_count': 2,
+            'out_ports_count': 1,
+        }
+        super().__init__(graph, mandatory_props, attrs)
+
+    def backend_attrs(self):
+        return ['batch_dims']
+
+    @staticmethod
+    def infer(node: Node):
+        node_name = node.soft_get('name', node.id)
+        connected_in_ports = [port for port in node.in_ports().values() if not port.disconnected()]
+        assert len(connected_in_ports) == 2, \
+            "Incorrect number of inputs for {} node".format(node_name)
+
+        data_shape = node.in_port(0).data.get_shape()
+        data_value = node.in_port(0).data.get_value()
+        indices_shape = node.in_port(1).data.get_shape()
+        indices_value = node.in_port(1).data.get_value()
+
+        batch_dims = 0
+        if node.has_valid('batch_dims'):
+            batch_dims = node.batch_dims
```

Comment:
You can add a default value to the Op constructor instead. And have an assert here `assert node.has_valid('batch_dims')`

Reply:
Fixed.